packages/csharp/PortingAssistant/PortingAssistant.Telemetry/Utils/LogUploadUtils.cs

```diff
@@ -58,48 +59,59 @@ TelemetryConfiguration telemetryConfiguration
                 {
-                    dynamic requestMetadata = new JObject();
-                    requestMetadata.version = "1.0";
-                    requestMetadata.service = telemetryConfiguration.ServiceName;
-                    requestMetadata.token = "12345678";
-                    requestMetadata.description = telemetryConfiguration.Description;
-
-                    dynamic log = new JObject();
-                    log.timestamp = DateTime.Now.ToString();
-                    log.logName = logName;
-                    var logDataInBytes = System.Text.Encoding.UTF8.GetBytes(logData);
-                    log.logData = System.Convert.ToBase64String(logDataInBytes);
-
-                    dynamic body = new JObject();
-                    body.requestMetadata = requestMetadata;
-                    body.log = log;
-
-                    var requestContent = new StringContent(body.ToString(Formatting.None), Encoding.UTF8, "application/json");
-
-                    var config = new TelemetryClientConfig
+                    awsCredentials = FallbackCredentialsFactory.GetCredentials();
+                    if (awsCredentials == null)
                     {
-                        RegionEndpoint = RegionEndpoint.GetBySystemName(region),
-                        MaxErrorRetry = 2,
-                        ServiceURL = telemetryConfiguration.InvokeUrl,
-                    };
-                    var client = new TelemetryClient(awsCredentials, config);
-                    var contentString = await requestContent.ReadAsStringAsync();
-                    var telemetryRequest = new TelemetryRequest(telemetryConfiguration.ServiceName, contentString);
-                    var telemetryResponse = await client.SendAsync(telemetryRequest);
-                    if (telemetryResponse.HttpStatusCode != HttpStatusCode.OK)
+                        Console.WriteLine("Invalid Credentials.");
+                        return false;
+                    }
+                }
+                else {
+                    var profileName = profile;
+                    if (!chain.TryGetAWSCredentials(profileName, out awsCredentials))
                     {
-                        Console.WriteLine("Http response failed with status code: " + telemetryResponse.HttpStatusCode.ToString());
+                        Console.WriteLine("Invalid Credentials.");
+                        return false;
                     }
-                    
-                    return telemetryResponse.HttpStatusCode == HttpStatusCode.OK;
                 }
-                Console.WriteLine("Invalid Credentials.");
-                return false;
-            }
+                var region = telemetryConfiguration.Region;
+                  dynamic requestMetadata = new JObject();
+                  requestMetadata.version = "1.0";
+                  requestMetadata.service = telemetryConfiguration.ServiceName;
+                  requestMetadata.token = "12345678";
```

Comment:
what is this token used for?

Reply:
It's supposed to be for verification but it has not been implemented on the api side and so we have been using a dummy value.